docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,691 @@
+                    "utxo_pos"  : 10001001,
+                    "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+                    "currency"  : "0000000000000000000000000000000000000000",
+                    "amount" : 100
+                }
+            }
+        ]
+    }
+}
+```
+
+Gets plasma network and Watcher status
+
+### HTTP Request
+
+`POST /status`
+
+### Request Body
+
+No parameters are required.
+
+<aside class="warning">
+The most critical function of the Watcher is to monitor the ChildChain and report dishonest activity. 
+The user must call the `/status` endpoint periodically to check. Any situation that requires the user to either exit or challenge an invalid exit will be included in the `byzantine_events` field.
+</aside>
+
+### Byzantine events
+All of the following events indicate byzantine behaviour and that the user should either exit or challenge.
+
+#### `invalid_exit`
+> An invalid_exit event
+
+```json
+{
+    "event": "invalid_exit",
+    "details": {
+        "eth_height"  : 3521678,
+        "utxo_pos"  : 10001001,
+        "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+        "currency"  : "0000000000000000000000000000000000000000",
+        "amount" : 100
+    }
+}
+```
+
+Indicates that an invalid exit is occurring. It should be challenged.
+
+
+#### `unchallenged_exit`
+> An unchallenged_exit event
+
+```json
+{
+    "event": "unchallenged_exit",
+    "details": {
+        "eth_height"  : 3521678,
+        "utxo_pos"  : 10001001,
+        "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+        "currency"  : "0000000000000000000000000000000000000000",
+        "amount" : 100
+    }
+}
+```
+
+Indicates that an invalid exit is dangerously close to finalization and hasn't been challenged. User should exit.
+
+
+#### `invalid_block`
+> An invalid_block event
+
+```json
+{
+    "event": "invalid_block",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+A block containing an invalid tx has been processed [are there other reasons a block can be invalid?]. User should exit.
+
+
+#### `block_withholding`
+> A block_withholding event
+
+```json
+{
+    "event": "block_withholding",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+The ChildChain is withholding a block whose hash has been published on the root chain. User should exit.
+
+#### `noncanonical_ife`
+> A noncanonical_ife event
+
+```json
+{
+    "event": "noncanonical_ife",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_input_index"  : 1,
+        "competing_blocknum"  : 10000,
+        "competing_tx_index"  : 10,
+        "competing_input_index"  : 0,
+    }
+}
+```
+
+An non-canonical in-flight exit has been started. It should be challenged. 
```

Comment:
1/ `non-canonical transaction in-flight exit` or `in-flight exit of a non-canonical...` is more precise. Ah I see we settled on `noncanonical_ife` as the name, I guess the abbreviation there is ok, but in the description a full, precise wording is probably clearer.
2/ the non-canonicity can be caused by a transaction, that isn't included in a block, so the tripplet `"competing_..."` won't cover all cases. The most visual scenario is when two IFEs for competing txs are started, but none is included in a block.

The part that is related to the competing tx being included should probably be optional, but pls double check this with the protocol specs

And it also digs into the issue what to serve in here and what to accept in the respective "information returning" endpoint - one that gets called to take action on Ethereum